hs_core/urls.py

```diff
@@ -4,36 +4,62 @@
+
+    # DEPRECATED: use GET /resource/ instead 
     url(r'^resourceList/$', views.resource_rest_api.ResourceList.as_view(),
-        name='list_resources'),
+        name='DEPRECATED_list_resources'),
 
-    url(r'^resource/$', views.resource_rest_api.ResourceCreate.as_view(),
+    url(r'^resource/$', views.resource_rest_api.ResourceListCreate.as_view(),
         name='create_resource'),
 
-    url(r'^resource/(?P<pk>[A-z0-9]+)/$', views.resource_rest_api.ResourceReadUpdateDelete.as_view(),
+    url(r'^resource/(?P<pk>[0-9a-f-]+)/$', views.resource_rest_api.ResourceReadUpdateDelete.as_view(),
         name='get_update_delete_resource'),
 
-    url(r'^resource/accessRules/(?P<pk>[A-z0-9]+)/$', views.resource_rest_api.AccessRulesUpdate.as_view(),
+    url(r'^resource/(?P<pk>[0-9a-f-]+)/access/$', views.resource_rest_api.AccessRulesUpdate.as_view(),
         name='update_access_rules'),
 
-    url(r'^sysmeta/(?P<pk>[A-z0-9]+)/$', views.resource_rest_api.SystemMetadataRetrieve.as_view(),
+    # DEPRECATED: use form above instead 
+    url(r'^resource/accessRules/(?P<pk>[0-9a-f-]+)/$', views.resource_rest_api.AccessRulesUpdate.as_view(),
+        name='DEPRECATED_update_access_rules'),
+
+    url(r'^resource/(?P<pk>[0-9a-f-]+)/sysmeta/$', views.resource_rest_api.SystemMetadataRetrieve.as_view(),
         name='get_system_metadata'),
 
-    url(r'^scimeta/(?P<pk>[A-z0-9]+)/$', views.resource_rest_api.ScienceMetadataRetrieveUpdate.as_view(),
+    # DEPRECATED: use form above instead 
+    url(r'^sysmeta/(?P<pk>[0-9a-f-]+)/$', views.resource_rest_api.SystemMetadataRetrieve.as_view(),
+        name='DEPRECATED_get_system_metadata'),
+
+    url(r'^resource/(?P<pk>[0-9a-f-]+)/scimeta/$', views.resource_rest_api.ScienceMetadataRetrieveUpdate.as_view(),
         name='get_update_science_metadata'),
 
-    url(r'^resource/(?P<pk>[A-z0-9]+)/files/$', views.resource_rest_api.ResourceFileCRUD.as_view(),
-        name='add_resource_file'),
+    # DEPRECATED: use form above instead
+    url(r'^scimeta/(?P<pk>[0-9a-f-]+)/$', views.resource_rest_api.ScienceMetadataRetrieveUpdate.as_view(),
+        name='DEPRECATED_get_update_science_metadata'),
+
+    # Unused. See ResourceFileListCreate. 
+    # url(r'^resource/(?P<pk>[0-9a-f-]+)/files/$', 
+    #     views.resource_rest_api.ResourceFileCRUD.as_view(),
+    #     name='add_resource_file'),
 
-    url(r'^resource/(?P<pk>[A-z0-9]+)/files/(?P<filename>[^/]+)/$',
-        views.resource_rest_api.ResourceFileCRUD.as_view(), name='get_update_delete_resource_file'),
+    url(r'^resource/(?P<pk>[0-9a-f-]+)/files/(?P<filename>[^/]+)/$',
+        views.resource_rest_api.ResourceFileCRUD.as_view(), 
+        name='get_update_delete_resource_file'),
 
-    url(r'^resource/(?P<pk>[A-z0-9]+)/file_list/$', views.resource_rest_api.ResourceFileList.as_view(),
+    url(r'^resource/(?P<pk>[0-9a-f-]+)/files/$', 
+        views.resource_rest_api.ResourceFileListCreate.as_view(),
         name='get_resource_file_list'),
```

Comment:
I moved the create function to here. Hence the name change. 
* The way it was done before, the list function was separate from the create function and create was in the CRUD class. 
* The proper way to do this is that List and Create are part of the same class (they share the same invocation pattern).

So, I moved them.